src/components/base/sw-banner/sw-banner.vue

```diff
@@ -1,392 +1,435 @@
 <template>
   <div
-    class="sw-alert"
-    :class="alertClasses"
-    role="alert"
+    class="sw-banner"
+    :class="bannerClasses"
+    role="banner"
   >
     <slot name="customIcon">
       <sw-icon
-        v-if="showIcon"
-        size="20px"
-        class="sw-alert__icon"
-        :name="alertIcon"
+        v-if="!hideIcon"
+        class="sw-banner__icon"
+        :name="bannerIcon"
         decorative
       />
     </slot>
-
     <div
-      class="sw-alert__body"
-      :class="alertBodyClasses"
+      class="sw-banner__body"
+      :class="bannerBodyClasses"
     >
       <div
         v-if="title"
-        class="sw-alert__title"
+        class="sw-banner__title"
       >
         {{ title }}
       </div>
-
-      <div class="sw-alert__message">
+      <div class="sw-banner__message">
         <slot />
       </div>
     </div>
-
     <div
       v-if="hasActionSlot"
-      class="sw-alert__actions"
+      class="sw-banner__actions"
     >
       <slot name="actions" />
     </div>
-
     <button
       v-if="closable"
-      class="sw-alert__close"
+      class="sw-banner__close"
       aria-label="Schließen"
       title="Schließen"
       @click.prevent="$emit('close', notificationIndex)"
     >
-      <sw-icon
-        name="small-default-x-line-medium"
-        small
-      />
+      <sw-icon name="regular-times-s" />
     </button>
   </div>
 </template>
 
-<script>
-import SwIcon from '../sw-icon/sw-icon.vue';
+<script lang="ts">
+import Vue from 'vue';
+import type { PropType } from 'vue';
+import SwIcon from '../sw-icon/sw-icon.vue'
+
+type AppearanceType = 'default' | 'notification' | 'system';
+type CssClasses = Array<string | Record<string, boolean>> | Record<string, boolean>;
+type NotificationType = 'info' | 'warning' | 'error' | 'success';
 
-export default {
+export default Vue.extend({
   name: 'SwBanner',
 
   components: {
     'sw-icon': SwIcon,
   },
 
   props: {
+    /**
+     * Change the variant of the banner
+     * @values info, warning, error, success
+     */
     variant: {
-      type: String,
-      default: 'info',
-      validValues: ['info', 'warning', 'error', 'success'],
-      validator(value) {
-        return ['info', 'warning', 'error', 'success'].includes(value);
-      },
+        type: String as PropType<NotificationType>,
+        required: false,
+        default: 'info',
+        validator(value: string): boolean {
+            return ['info', 'warning', 'error', 'success'].includes(value);
```

Comment:
Error should be named danger.

Reply:
I talked to Dennis. Error is now Critical.